File: backend/src/main/java/edu/emagazine/EmagazineApplication.java
package edu.emagazine;

import com.mongodb.MongoClient;
import com.mongodb.MongoClientURI;
import java.util.HashSet;
import java.util.Set;
import javax.ws.rs.ApplicationPath;
import javax.ws.rs.core.Application;

@ApplicationPath("api")
public class EmagazineApplication extends Application {

  public EmagazineApplication() {
  }

  /**
   *
   * @return entry point http://localhost:8085/api/
   */
  @Override
  public Set<Object> getSingletons() {
    HashSet<Object> set = new HashSet<Object>();
    //DataBase connection
    MongoClientURI uri = new MongoClientURI(
      "mongodb+srv://AdminEx:AdminEx@clusterex-cya5a.mongodb.net/test?retryWrites=true&w=majority");
    MongoClient mongoClient = new MongoClient(uri);

    set.add(new Emagazine(mongoClient));
    return set;
  }
}

File: backend/src/main/java/edu/emagazine/EmagazineApplication.kt
package edu.emagazine

import com.mongodb.MongoClient
import com.mongodb.MongoClientURI
import java.util.HashSet
import javax.ws.rs.ApplicationPath
import javax.ws.rs.core.Application

@ApplicationPath("api")
class EmagazineApplication : Application() {

    /**
     *
     * @return entry point http://localhost:8085/api/
     */
    override fun getSingletons(): Set<Any> {
        val set = HashSet<Any>()
        //DataBase connection
        val uri = MongoClientURI(
                "mongodb+srv://AdminEx:AdminEx@clusterex-cya5a.mongodb.net/test?retryWrites=true&w=majority")
        val mongoClient = MongoClient(uri)

        set.add(Emagazine(mongoClient))
        return set
    }
}

